reflex/config.py

```diff
@@ -274,6 +291,34 @@ def get_event_namespace(self) -> str | None:
         event_url = constants.Endpoint.EVENT.get_url()
         return urllib.parse.urlsplit(event_url).path
 
+    def _replace_defaults(self, **kwargs):
+        """Replace formatted defaults when the caller provides updates.
+
+        Args:
+            **kwargs: The kwargs passed to the config or from the env.
+        """
+        if "api_url" not in self._non_default_attributes and "backend_port" in kwargs:
+            self.api_url = f"http://localhost:{kwargs['backend_port']}"
+
+        if (
+            "deploy_url" not in self._non_default_attributes
+            and "frontend_port" in kwargs
+        ):
+            self.deploy_url = f"http://localhost:{kwargs['frontend_port']}"
```

Comment:
I don't think the `deploy_url` should ever be localhost. We only use this currently for the NextJS sitemap for indexing websites

Reply:
The default `deploy_url` is already "localhost" in the `Config`... what should it be?

Reply:
I guess `localhost` is accurate in dev mode, it's just currently only used for the sitemap so it's a bit meaningless.